pkg/server/cloud/gce/metadata.go

```diff
@@ -80,3 +81,31 @@ func getZoneFromMetadata(c *metadata.Client) (string, error) {
 	}
 	return extractZoneFromResponse(s)
 }
+
+func getGKEMetadata() map[string]string {
+	if !metadata.OnGCE() {
+		return nil
+	}
+	keys := []string{
+		"cluster-location",
+		"cluster-name",
+		"cluster-uid",
+	}
+	c := newMetadataClient()
+	gkeMD := make(map[string]string)
+	for _, k := range keys {
+		url := "instance/attributes/" + k
+		val, err := getMetadataTrimmed(c, url)
+		if err != nil {
+			// Pretty sure we can continue in this case. The cluster
+			// will still function, if super important, daemonSets
+			// might not run on our virtual node but we are running
+			// them mostly for apprarances sake.  They should be
+			// patched to not run on our kip nodes.
+			klog.Warningln("unable to retrieve gke metadata key", k)
+			continue
+		}
+		gkeMD[k] = val
+	}
+	return gkeMD
+}
```

Comment:
Do we handle the case when Kip is running outside of GCE (minikube or on-prem)? I guess it'll just log that the metadata service is not available and move on?

Reply:
I don't think we should take care of that case at this time.  If we're not inside GCE, we'll assume we're not on GKE.  If that assumption doesn't hold up we can make changes.